src/internal/zod.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+// No need to bundle the full zod library since we only use the typeNames
+// to identify the schemas.
+
+import type {
+	EnumLike,
+	ZodAny as TrueZodAny,
+	ZodArray as TrueZodArray,
+	ZodBigInt as TrueZodBigInt,
+	ZodBoolean as TrueZodBoolean,
+	ZodBranded as TrueZodBranded,
+	ZodCatch as TrueZodCatch,
+	ZodDate as TrueZodDate,
+	ZodDefault as TrueZodDefault,
+	ZodDiscriminatedUnion as TrueZodDiscriminatedUnion,
+	ZodEffects as TrueZodEffects,
+	ZodEnum as TrueZodEnum,
+	ZodFunction as TrueZodFunction,
+	ZodIntersection as TrueZodIntersection,
+	ZodLazy as TrueZodLazy,
+	ZodLiteral as TrueZodLiteral,
+	ZodMap as TrueZodMap,
+	ZodNaN as TrueZodNaN,
+	ZodNativeEnum as TrueZodNativeEnum,
+	ZodNever as TrueZodNever,
+	ZodNull as TrueZodNull,
+	ZodNullable as TrueZodNullable,
+	ZodNumber as TrueZodNumber,
+	ZodObject as TrueZodObject,
+	ZodOptional as TrueZodOptional,
+	ZodPipeline as TrueZodPipeline,
+	ZodPromise as TrueZodPromise,
+	ZodRecord as TrueZodRecord,
+	ZodSet as TrueZodSet,
+	ZodString as TrueZodString,
+	ZodSymbol as TrueZodSymbol,
+	ZodTuple as TrueZodTuple,
+	ZodUndefined as TrueZodUndefined,
+	ZodUnion as TrueZodUnion,
+	ZodUnknown as TrueZodUnknown,
+	ZodVoid as TrueZodVoid,
+	ZodDiscriminatedUnionOption,
+	ZodRawShape,
+	ZodTypeAny,
+} from 'zod';
+
+// Forces typescript to cast our fake constructors as a true zod type
+const castAs = <T extends ZodTypeAny>(value: string) => {
+	const fake = () => {};
+	Object.defineProperty(fake, 'name', { value });
```

Comment:
To "mark" this property

```suggestion
	Object.defineProperty(fake, '__zod-fixture_name', { value });
```

Reply:
This has to be the `name` property of the function in order to act the same way as a normal Zod constructor.

https://github.com/timdeschryver/zod-fixture/blob/d658e549da766e88a406e91ce4fb5d3cde65260d/src/transformer/transformer.ts#L49C1-L51C38
```ts
			const generaterType = isConstructor
				? generator.schema.name // <---- this part right here
				: generator.schema._def.typeName;
```

Technically the same as:
```ts
const castAs = <T extends ZodTypeAny>(func: unknown) => func as T

export const ZodString = castAs<TrueZodString>(function ZodString(){})
```
but when it gets compiled, it gets rewritten as
```ts
export const ZodString = castAs<TrueZodString>(function ZodString2(){})
```